dlt/common/configuration/specs/pluggable_run_context.py

```diff
@@ -206,7 +210,10 @@ def reload(
 
     def reload_providers(self) -> None:
         self.providers = ConfigProvidersContainer(self.context.initial_providers())
-        self.providers.add_extras()
+        # Invalidate any cached configuration on the context so it re-resolves using new providers
+        self.context.reset_config()
```

Comment:
I looked into the code and instead of adding new `reset_config` method you can just call
`self.initialize_runtime()` which will reset the config and also apply new settings to runtime (as define by contecxt provider)